scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,218 @@
+    if os.path.isfile(e2e_image_tag_override):
+        with open(e2e_image_tag_override, 'r') as tag:
+            e2e_image_tag = tag.read()
+
+    docker_env_ignore = [
+      'GOOGLE_APPLICATION_CREDENTIALS',
+      'GOROOT',
+      'HOME',
+      'PATH',
+      'PWD',
+      'WORKSPACE'
+    ]
+
+    docker_extra_args = []
+    if (os.environ.get('JENKINS_ENABLE_DOCKER_IN_DOCKER') and
+            os.environ.get('JENKINS_ENABLE_DOCKER_IN_DOCKER').lower() == 'y'):
+        docker_extra_args.extend([
+          '-v', '/var/run/docker.sock:/var/run/docker.sock',
+          '-v', '%s:/go/src/k8s.io/kubernetes' % repo,
+          '-e', 'REPO_DIR=%s' % repo,
+          '-e', 'HOST_ARTIFACTS_DIR=%s' % artifacts
+        ])
+
+    if (os.environ.get('JENKINS_USE_LOCAL_BINARIES') and
+            os.environ.get('JENKINS_USE_LOCAL_BINARIES').lower() == 'y'):
+        docker_extra_args.extend([
+          '-v', '%s/_output":/workspace/_output:ro' % workspace
+        ])
+
+    if 'KUBE_E2E_RUNNER' in os.environ:
+        docker_extra_args.extend([
+          '--entrypoint=%s', os.environ.get('KUBE_E2E_RUNNER')
+        ])
+
+    # exec
+
+    print 'Starting %s...' % CONTAINER
+
+    cmd = ['docker', 'run', '--rm',
+      '--name=%s' % CONTAINER,
+      '-v', '%s/_artifacts":/workspace/_artifacts' % workspace,
+      '-v', '/etc/localtime:/etc/localtime:ro'
+    ]
+
+    gce_ssh = '/workspace/.ssh/google_compute_engine'
+    gce_pub = '%s.pub' % gce_ssh
+    service = '/service-account.json'
+
+    cmd.extend(['-v', '%s:%s:ro' % (args.gce_ssh, gce_ssh)])
+    cmd.extend(['-v', '%s:%s:ro' % (args.gce_pub, gce_pub)])
+    cmd.extend(['-v', '%s:%s:ro' % (args.service_account, service)])
+
+    cmd.extend(['-e', 'JENKINS_GCE_SSH_PRIVATE_KEY_FILE=%s' % gce_ssh])
+    cmd.extend(['-e', 'JENKINS_GCE_SSH_PUBLIC_KEY_FILE=%s' % gce_pub])
+    cmd.extend(['-e', 'GOOGLE_APPLICATION_CREDENTIALS=%s' % service])
+
+    # Boilerplate envs
+
+    # Skip gcloud update checking
+    cmd.extend(['-e', 'CLOUDSDK_COMPONENT_MANAGER_DISABLE_UPDATE_CHECK=\'true\''])
+    # Use default component update behavior
+    cmd.extend(['-e', 'CLOUDSDK_EXPERIMENTAL_FAST_COMPONENT_UPDATE=\'false\''])
+
+    cmd.extend(['-e', 'E2E_UP=%s' % args.e2e_up])
+    cmd.extend(['-e', 'E2E_TEST=%s' % args.e2e_test])
+    cmd.extend(['-e', 'E2E_DOWN=%s' % args.e2e_down])
+
+    cmd.extend(['-e', 'E2E_NAME=%s' % args.e2e_name])
+    
+    # AWS
+    cmd.extend(['-e', 'KUBE_AWS_INSTANCE_PREFIX=%s' % args.e2e_name])
+
+    # GCE
+    cmd.extend(['-e', 'INSTANCE_PREFIX=%s' % args.e2e_name])
+    cmd.extend(['-e', 'KUBE_GCE_NETWORK=%s' % args.e2e_name])
+    cmd.extend(['-e', 'KUBE_GCE_INSTANCE_PREFIX=%s' % args.e2e_name])
+
+    # GKE
+    cmd.extend(['-e', 'CLUSTER_NAME=%s' % args.e2e_name])
+    cmd.extend(['-e', 'KUBE_GKE_NETWORK=%s' % args.e2e_name])
+
+    for key, value in os.environ.items():
+        if key not in docker_env_ignore:
+            cmd.extend(['-e', '%s=%s' % (key, value)])
```

Comment:
I'm pretty sure this ordering regresses how things work: specifically if my environment has `export INSTANCE_PREFIX=foo` it ill now do the following:

```
docker run -e INSTANCE_PREFIX=bootstrap-e2e -e INSTANCE_PREFIX=foo
```

Previously the foo.sh would override INSTANCE_PREFIX=foo with INSTANCE_PREFXI=bootstrap-e2e and then we would only do the following:
```
env > env.list
docker run --env-file=env.list  # aka only one -e INSTANCE_PREFIX=bootstrape2e
```

Is this change in behavior intentional?